### PR TITLE
Handle more pod/container error cases

### PR DESCRIPTION
**Pull request by joshwget**

### CODE REVIEW

File: sync/reconcile_pod.go [hunk 1]
```diff
@@ -36,12 +40,14 @@ func reconcilePod(clientset *kubernetes.Clientset, watchClient *watch.Client, de
 				if err := deletePod(clientset, watchClient, namespace, podName); err != nil {
 					return nil, err
 				}
+			} else {
+				break
```
Comment (joshwget) [inline sync/reconcile_pod.go]: Somewhat not relevant to this PR, but this piece is needed for `createPod` to not get always get called 5 times.

File: sync/reconcile_pod.go [hunk 2]
```diff
@@ -12,11 +14,13 @@ import (
...
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/pkg/api/v1"
-	"strings"
 )
 
 const (
-	waitTime = time.Second
+	waitTime                 = time.Second
+	timeoutDuration          = 5 * time.Minute
+	imagePullBackOffReason   = "ImagePullBackOff"
```
Comment (joshwget) [inline sync/reconcile_pod.go]: These are the only two currently considered errors.

File: sync/reconcile_pod.go (hunk 1 above)
Comment (ibuildthecloud) [reply sync/reconcile_pod.go]: Nice!

File: sync/reconcile_pod.go [hunk 3]
```diff
@@ -55,19 +61,24 @@ func waitForCacheToContainPod(watchClient *watch.Client, namespace, podName stri
...
 			}
 
-			currentStatusMessage := getPodStatusMessage(existingPod)
+			currentStatusMessage, err := getPodStatusMessage(existingPod)
+			if err != nil {
+				return nil, err
+			}
+
 			if currentStatusMessage != statusMessage {
+				// TODO: does deploymentSyncRequest need to be set for transitioning message?
```
Comment (ibuildthecloud) [inline sync/reconcile_pod.go]: No, a transitioning message doesn't need to have a body

**Comment by ibuildthecloud** [comment]
LGTM

File: sync/reconcile_pod.go (hunk 3 above)
Comment (joshwget) [reply sync/reconcile_pod.go]: I'll remove this in a separate PR, it's been this way for a while so it's probably harmless.

**Pull request merged**